Automattic/sensei
# Add tracks event in Sensei tour component and track Course and Lesson onboarding tour steps

Resolves https://github.com/Automattic/sensei/issues/7467

## Proposed Changes
We've added tracking functionality to track when a user opens a tour step. This is implemented in the main SenseiTourKit component, so for each subsequent tour, we'll just need to pass the ID.

We've added its usage in Course and Lesson tour.

We should also probably need to update the info https://senseilms.com/documentation/what-data-does-sensei-track/ because we're tracking a new data. Addding the "Needs Documentation" label to this PR for that purpose.

## Testing Instructions


1. Make sure your local env is set to `development` (to enable the feature flag)
2.  Make sure your tracking is enabled (Sensei LMS -> Settings -> General -> Enable usage tracking )
3. Create a new Course or open an existing one in the editor (You'll need a new user in case you've closed the tour before, or you can remove the value of the usermeta `sensei_tours`)
4. Complete the Course creation modal and make sure you have a Course Outline block in editor, make sure the Tour has appeared
5. Now go through each of the Tour steps
6. Now create a new lesson/open an existing one (same logic as above in case if have closed the tour before)
7. Make sure the lesson creation modal is closed and a quiz block is there in the editor, make sure you see the lesson tour
8. Now take each step of the tour
9. Now look in tracks for `sensei_course_outline_onboarding_step_complete` event, check the values and especially if the `step` property is there
10. Also look for the `sensei_lesson_quiz_onboarding_step_complete` event similarly to above.

## Pre-Merge Checklist
